app/services/vot/parser.rb

```diff
@@ -0,0 +1,96 @@
+module Vot
+  class Parser
+    class ParseException < StandardError; end
+    SUPPORTED_COMPONENT_VALUE_REQUIREMENTS = [
+      :aal2?,
+      :phishing_resistant?,
+      :hspd12?,
+      :identity_proofing?,
+      :biometric_comparison?,
+      :ialmax?,
+    ].freeze
+    Result = Data.define(:vector_of_trust, *SUPPORTED_COMPONENT_VALUE_REQUIREMENTS)
+
+    attr_reader :vector_of_trust
+
+    def initialize(vector_of_trust)
+      @vector_of_trust = vector_of_trust
+    end
+
+    def parse
+      initial_components = map_initial_vector_of_trust_componets_to_component_values
+      expand_components_with_initial_components(initial_components)
+    end
+
+    def parse_acr
+      initial_components = map_initial_acr_values_to_component_values
+      expand_components_with_initial_components(initial_components)
+    end
+
+    private
+
+    def map_initial_vector_of_trust_componets_to_component_values
+      vector_of_trust.split('.').map do |component_value_name|
+        component_value = SupportedComponentValues::MAPPED_BY_NAME[component_value_name]
+        if component_value.nil?
+          raise_unsupported_component_exception(component_value_name)
+        end
+        component_value
+      end
+    end
+
+    def map_initial_acr_values_to_component_values
+      vector_of_trust.split(' ').map do |component_value_name|
```

Comment:
I'm not sure I totally follow what we're doing with the legacy ACR values here, they're not valid for VoT?

Reply:
Vectors of trust take precedence over ACR values. But if there is not vector of trust then we use ACR values. The format is similar which means we can have a utility that handles both and responds with a structured object to describe the identity/authentication transaction we need to perform.